src/common/tracer.cc

```diff
@@ -1,86 +1,54 @@
 // -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
 // vim: ts=8 sw=2 smarttab
 
+#include "common/ceph_context.h"
+#include "global/global_context.h"
 #include "tracer.h"
-#include <arpa/inet.h>
-#include <yaml-cpp/yaml.h>
-#ifdef __linux__
-#include <linux/types.h>
-#else
-typedef int64_t __s64;
-#endif
 
-#include "common/debug.h"
+#ifdef HAVE_JAEGER
 
-#define dout_context g_ceph_context
-#define dout_subsys ceph_subsys_osd
-#undef dout_prefix
-#define dout_prefix *_dout << "jaeger_tracing "
+namespace tracing {
 
-namespace jaeger_tracing {
+const std::shared_ptr<opentracing::Tracer> Tracer::noop_tracer = opentracing::MakeNoopTracer();
 
-  std::shared_ptr<opentracing::v3::Tracer> tracer = nullptr;
+Tracer::Tracer(jaegertracing::Config& conf):open_tracer(jaegertracing::Tracer::make(conf)) {}
 
-  void init_tracer(const char* tracer_name) {
-    if (!tracer) {
-	YAML::Node yaml;
-	try{
-	  yaml = YAML::LoadFile("../src/jaegertracing/config.yml");
-	dout(3) << "yaml loaded" << yaml << dendl;
-	}
-	catch(std::exception &e) {
-	  dout(3) << "failed to load yaml file using default config" << dendl;
-	  auto yaml_config = R"cfg(
-disabled: false
-reporter:
-    logSpans: false
-    queueSize: 100
-    bufferFlushInterval: 10
-sampler:
-    type: const
-    param: 1
-headers:
-    jaegerDebugHeader: debug-id
-    jaegerBaggageHeader: baggage
-    TraceContextHeaderName: trace-id
-baggage_restrictions:
-    denyBaggageOnInitializationFailure: false
-    refreshInterval: 60
-)cfg";
-	yaml = YAML::Load(yaml_config);
-	dout(3) << "yaml loaded" << yaml << dendl;
-      }
-      static auto configuration = jaegertracing::Config::parse(yaml);
-      tracer = jaegertracing::Tracer::make( tracer_name, configuration,
-	    jaegertracing::logging::consoleLogger());
-      dout(3) << "tracer_jaeger init successful" << dendl;
-    }
-    //incase of stale tracer, configure with a new global tracer
-    if (opentracing::Tracer::Global() != tracer) {
-      opentracing::Tracer::InitGlobal(
-	  std::static_pointer_cast<opentracing::Tracer>(tracer));
-    }
-  }
+Tracer::Tracer(opentracing::string_view service_name) {
```

Comment:
in which case are we using this constructor?
also, why do we need?
```cpp
Tracer(jaegertracing::Config& conf)
```

Reply:
> in which case are we using this constructor?

currently, it is unused. in case one would like to create a tracer with custom service_name it could be with that c'tor.  the second option is what goes in the osd - since it's a global variable, we use the default c'tor and init the tracer with the service name during run time.


> also, why do we need?
> 
> ```c++
> Tracer(jaegertracing::Config& conf)
> ```


it was originally made for the rgw tracer. in order to avoid the construction of the `jagertracing::Config` object in each thread.

Reply:
maybe add this ctor:
```
Tracer::Tracer(opentracing::string_view service_name, jaegertracing::Config& conf)
```
to replace `Tracer(jaegertracing::Config& conf)` and `Tracer(opentracing::string_view service_name)`
so the rgw code looks more like the osd one